util/src_util/pt_auth_as.c

```diff
@@ -0,0 +1,1204 @@
+int num_frames_1K = 1000;
+int num_frames_100 = 100;
+
+void Write_To_File(uint16_t enc_frame_len, float total_time, char* test_name, int num_frames, int reset)
+{
+    if(reset == 1)
+    {
+        if(access("PERFORMANCE_RESULTS_AAS.csv", F_OK) == 0)
+        {
+            int deleted = remove("PERFORMANCE_RESULTS_AAS.csv");
+            if(deleted){printf("ERROR Deleting File!\n");}
+        }
+    }
+
+    FILE *fp = NULL;
+    fp = fopen("PERFORMANCE_RESULTS_AAS.csv", "a");
+    if (fp != NULL)
+    {   
+        if(reset ==1) fprintf(fp, "Name of Test,Frames Sent,Bytes per Frame,Total Time,Mbps\n");
+        fprintf(fp, "%s,%d,%d,%f,%f\n", test_name, num_frames, enc_frame_len, total_time, (((enc_frame_len * 8 * num_frames)/total_time)/1024/1024));
+    }
+    fclose(fp);
+    
+}
+
+double Apply_Security_Loop(uint8_t *frame, int frame_length, uint8_t *enc_frame, uint16_t *enc_frame_len, int num_loops)
+{
+    struct timespec begin, end;
+    double total_time;
+    total_time = 0.0;
+
+    frame = frame;
+    frame_length = frame_length;
+    enc_frame_len = enc_frame_len;
+
+    int32_t status = CRYPTO_LIB_SUCCESS;
+
+    for(int i = 0; i < num_loops; i++)
+    {
+        printf("LOOP NUMBER: %d\n", i+1);
+        clock_gettime(CLOCK_REALTIME, &begin);
+        status = Crypto_TC_ApplySecurity(frame, frame_length, &enc_frame, enc_frame_len);
+        clock_gettime(CLOCK_REALTIME, &end);
+        free(enc_frame);
+
+        long seconds = end.tv_sec - begin.tv_sec;
+        long nanoseconds = end.tv_nsec - begin.tv_nsec;
+        double elapsed = seconds + nanoseconds*1e-9;
+
+        if (status != CRYPTO_LIB_SUCCESS)
+        {
+            total_time = -1.0;
+            printf("ERROR: %d\n", status);
+            break;
+        }
+
+        total_time += elapsed;
+    }
+    return total_time;
+}
+
+UTEST(PERFORMANCE, LSA_LIBG_AUTH_SHORT_100)
+{
+    // Setup & Initialize CryptoLib
+    Crypto_Config_CryptoLib(SADB_TYPE_INMEMORY, CRYPTOGRAPHY_TYPE_LIBGCRYPT, CRYPTO_TC_CREATE_FECF_TRUE, TC_PROCESS_SDLS_PDUS_FALSE, TC_NO_PUS_HDR,
+                            TC_IGNORE_SA_STATE_FALSE, TC_IGNORE_ANTI_REPLAY_TRUE, TC_UNIQUE_SA_PER_MAP_ID_FALSE,
+                            TC_CHECK_FECF_TRUE, 0x3F, SA_INCREMENT_NONTRANSMITTED_IV_TRUE);
+    
+    Crypto_Config_Add_Gvcid_Managed_Parameter(0, 0x002C, 1, TC_HAS_FECF, TC_NO_SEGMENT_HDRS, 1024);
+    int32_t status = Crypto_Init();
+    Crypto_Init();
+    char* data_h = "202C046100ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEF0123456789ABCDEFF01C";
+    char* data_b = NULL;
+    int data_l = 0;
+
+    uint8_t* ptr_enc_frame = NULL;
+    uint16_t enc_frame_len = 0;
+    SadbRoutine sadb_routine = get_sadb_routine_inmemory();
+
+    hex_conversion(data_h, &data_b, &data_l);
+
+    SecurityAssociation_t* test_association = malloc(sizeof(SecurityAssociation_t) * sizeof(uint8_t));
```

Comment:
Does `test_association` get freed anywhere? May apply to multiple tests.

Reply:
Not within the unit tests, no